src/Features/CSharp/Portable/Completion/CompletionProviders/AwaitCompletionProvider.cs

```diff
@@ -50,6 +53,92 @@ private protected override int GetSpanStart(SyntaxNode declaration)
         }
 
         private protected override SyntaxNode? GetAsyncSupportingDeclaration(SyntaxToken token)
-            => token.GetAncestor(node => node.IsAsyncSupportingFunctionSyntax());
+        {
+            var declaration = token.GetAncestor(node => node.IsAsyncSupportingFunctionSyntax());
+            // In a case like
+            //   someTask.$$
+            //   await Test();
+            // someTask.await Test() is parsed as a local function statement.
+            // We skip this and look further up in the hierarchy.
+            if (declaration is LocalFunctionStatementSyntax localFunction && localFunction.ReturnType.ChildTokens().Contains(token))
+            {
+                return localFunction.Parent?.FirstAncestorOrSelf<SyntaxNode>(node => node.IsAsyncSupportingFunctionSyntax());
+            }
+
+            return declaration;
+        }
```

Comment:
dumb question.  why can't you just use the GetExpressionToPlaceAwaitInFrontOf function to determine things.  I'ts not clear to me (yet) why you need this declaration.

Reply:
This is the declaration where the `async` is placed, if needed. It's basically asking for the surrounding method of the token. In the odd case described, the token is within a parsed method declaration, but we assume it is a member access. Without this special case handling an `async` would be placed somewhere right here.

I tried to pass another token (the one passed is the one at the cursor position), but all tokens at hand are within the LocalFunctionSyntax.

It is a bad hack here and hard to describe why it's needed, but I couldn't figure out something better yet.

Reply:
See also #56196